merge_command.go

```diff
@@ -1,17 +1,34 @@
 package main
 
-import "strings"
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
 
 const mergingLabel = "merging"
 
 func isMergeCommand(comment string) bool {
 	return strings.TrimSpace(comment) == "!merge"
 }
 
-func handleMergeCommand(issueComment IssueComment, issues Issues) Response {
-	err := addLabel(issueComment.Repository, issueComment.IssueNumber, mergingLabel, issues)
+func handleMergeCommand(issueComment IssueComment, issues Issues, pullRequests PullRequests) Response {
+	errResp := addLabel(issueComment.Repository, issueComment.IssueNumber, mergingLabel, issues)
+	if errResp != nil {
+		return errResp
+	}
+	pr, errResp := getPR(issueComment, pullRequests)
+	if errResp != nil {
+		return errResp
+	} else if *pr.Merged {
+		return SuccessResponse{}
+	} else if !*pr.Mergeable {
+		return SuccessResponse{}
```

Comment:
Yes. In some situations the PR simply isn't currently mergeable and so the webhook says "fine" and gives up. And as I explain in the PR description this is okay, because we keep trying to merge PRs that have the 'merging' label and we won't remove the label until we've actually merged the PR. So as long as we're listening to all the correct events (i.e. have correct webhooks set up, which will not be the case with this PR) then we can guarantee that everything will get merged eventually.